addon/services/ajax.js

```diff
@@ -113,9 +113,9 @@ export default Ember.Service.extend({
         );
 
         if (response instanceof AjaxError) {
-          reject(response);
+          Ember.run.join(null, reject, response);
```

Comment:
Ok, @alexlafroscia and I will write a few tests. If we're not able to replicate the problem then we'll just close this issue. Thank you for bringing this to my attention.